TombEngine/Objects/TR2/Entity/tr2_spear_guardian.cpp

```diff
@@ -17,34 +18,102 @@ using namespace TEN::Math;
 
 namespace TEN::Entities::Creatures::TR2
 {
+	constexpr auto SPEAR_GUARDIAN_WALK_ANGLE_RATE_MAX = ANGLE(3.0f);
+	constexpr auto SPEAR_GUARDIAN_RUN_ANGLE_RATE_MAX = ANGLE(5.0f);
+
 	const auto SpearBiteLeft  = BiteInfo(Vector3(0.0f, 0.0f, 920.0f), 11);
 	const auto SpearBiteRight = BiteInfo(Vector3(0.0f, 0.0f, 920.0f), 18);
 
-	// TODO
+	constexpr auto SPEAR_GUARDIAN_SLASH_RANGE = SQUARE(SECTOR(1));
+	constexpr auto SPEAR_GUARDIAN_DOUBLE_ATTACK_RANGE = SQUARE(SECTOR(1));
+	constexpr auto SPEAR_GUARDIAN_DOUBLE_ATTACK_WALK_RANGE = SQUARE(SECTOR(1.5f));
+	constexpr auto SPEAR_GUARDIAN_RUN_RANGE = SQUARE(SECTOR(3));
+	constexpr auto SPEAR_GUARDIAN_ATTACK_RANGE = SQUARE(SECTOR(2));
+
+	constexpr auto SPEAR_GUARDIAN_BASIC_DAMAGE = 75;
+	constexpr auto SPEAR_GUARDIAN_POWERFUL_DAMAGE = 120;
+	constexpr auto SPEAR_GUARDIAN_SWAPMESH_TIME = 3;
+
 	enum SpearGuardianState
 	{
-
+		// No state 0.
+		SPEAR_GUARDIAN_STATE_STOP = 1,
```

Comment:
Same, maybe **Idle** instead of **Stop**?

Reply:
Done